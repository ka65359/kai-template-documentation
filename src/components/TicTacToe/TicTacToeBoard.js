import React from "react";
const Square = ({ onClick, value }) => {
  return (
    <button className="square" onClick={onClick}>
      {value}
    </button>
  );
};

const TicTacToeBoard = ({ squares, onClick }) => {
  const renderSquare = (i) => {
    return <Square value={squares[i]} onClick={() => onClick(i)} />;
  };

  return (
    <div>
      <div className="board-row">
        {renderSquare(0)}
        {renderSquare(1)}
        {renderSquare(2)}
      </div>
      <div className="board-row">
        {renderSquare(3)}
        {renderSquare(4)}
        {renderSquare(5)}
      </div>
      <div className="board-row">
        {renderSquare(6)}
        {renderSquare(7)}
        {renderSquare(8)}
      </div>
    </div>
  );
};

export default TicTacToeBoard;
