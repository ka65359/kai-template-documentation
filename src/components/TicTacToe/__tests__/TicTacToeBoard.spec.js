import React from "react";
import { shallow } from "enzyme";
import TicTacToeBoard from "../TicTacToeBoard";

describe("TicTacToeBoard() tests", () => {
  let onClickSpy;
  let squares;

  beforeEach(() => {
    onClickSpy = jest.fn();
    squares = [];
    for (let i = 0; i < 9; i++) {
      squares.push(i + 1);
    }
  });

  afterEach(() => {
    onClickSpy.mockReset();
    onClickSpy = null;
    squares = null;
  });

  it("should render correctly in mode", () => {
    const component = shallow(<TicTacToeBoard onClick squares />);

    expect(component).toMatchSnapshot();
  });
});
