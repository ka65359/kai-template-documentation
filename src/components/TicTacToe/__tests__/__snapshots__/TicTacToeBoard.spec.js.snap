// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicTacToeBoard() tests should render correctly in mode 1`] = `ShallowWrapper {}`;
